src/main/java/org/candlepin/subscriptions/tally/billing/TallySummaryMessageConsumer.java

```diff
@@ -71,8 +74,10 @@ public TallySummaryMessageConsumer(
       topics = "#{__listener.topic}",
       containerFactory = "billingProducerKafkaTallySummaryListenerContainerFactory")
   @Transactional
-  public void receive(TallySummary tallySummary) {
-    log.debug("Tally Summary received. Producing billable usage.}");
+  public void receive(
+      @Payload TallySummary tallySummary,
+      @Header(name = KafkaHeaders.RECEIVED_MESSAGE_KEY, required = false) String kafkaMessageKey) {
+    log.debug("Tally Summary received w/ key={}. Producing billable usage.", kafkaMessageKey);
```

Comment:
By not requiring the key, that just means that it wouldn't guarantee the message is processed by the same consumer, right?  Other than not reducing the risk of concurrency bugs, is there any other drawback to that?  I'm assuming we're doing required=false for now so we don't get rejected messages....but would we want to toggle that to required=true in a future release?

Reply:
`required = false` just allows `null` to be passed.

Per the javadocs:

> Default is true, leading to an exception if the header is missing. Switch this to false if you prefer a null value in case of a header missing.

With this, if we have some pending messages during rollout, then they won't cause exceptions.